services/app-web/src/pages/StateSubmission/ContractDetails/ContractDetails.test.tsx

```diff
@@ -187,19 +183,13 @@ describe('ContractDetails', () => {
         })
 
         it('displays correct form fields for federal authorities with CHIP only contract', async () => {
-            vi.spyOn(
-                useHealthPlanPackageForm,
-                'useHealthPlanPackageForm'
-            ).mockImplementation(() => {
-                return {
-                    createDraft: vi.fn(),
-                    updateDraft: mockUpdateDraftFn,
-                    showPageErrorMessage: false,
-                    draftSubmission: {
-                        ...mockContractAndRatesDraft(),
-                        populationCovered: 'CHIP',
-                    },
-                }
+            const draftContract = mockContractPackageUnlockedWithUnlockedType()
+            draftContract.draftRevision!.formData.populationCovered = 'CHIP'
+            vi.spyOn(useContractForm, 'useContractForm').mockReturnValue({
```

Comment:
I'm interested in us discussing this pattern in a broader testing patterns discussion. I don't understand the benefits to mocking out a bunch of the logic in useContractForm vs. the pattern we've been using mostly of mocking the fetchContract response at the ApolloClient layer

Reply:
That said, this is obviously adapting a pattern we already had so I think this PR doesn't need to wait on that discussion

Reply:
I wasn't sure if there were strong opinions on this but like the idea of a wider discussion. For this I used the existing pattern for this file to save time

Reply:
FYI there's nothing intentional going on here in the code from before. Down to talk in RSP but I don't consider anything with useHealthPlanPackageForm a pattern to copy, it was hurridedly created trying to de-scope linked rates work, it was not planned for, and it  was always meant to be throwaway. I want us to create patterns and be more intentional now since we have actual sprint time dedicated to Contract API.

Reply:
I still need to make a separate cleanup ticket for the unlockedContract mock in test. I can add in refactoring the contract details test to use the updated pattern as well